gdal/port/cpl_json.cpp

```diff
@@ -0,0 +1,1154 @@
+ * @param  bDefault   Default value.
+ * @return            Boolean value.
+ *
+ * @since GDAL 2.3
+ */
+bool CPLJSONObject::GetBool(const char *pszName, bool bDefault) const
+{
+    if( nullptr == pszName )
+        return bDefault;
+    CPLJSONObject object = GetObj( pszName );
+    return object.ToBool( bDefault );
+}
+
+/**
+ * \brief Get json object children.
+ *
+ * This is not an array [], but list {}. Any modification of children will not store in json document. This function is useful when keys is not know and need to iterate over json object items and get keys and values.
+ *
+ * @return Array of CPLJSONObject pointers. The caller must free this array using CPLJSONObject::DestroyJSONObjectList static function.
+ *
+ * @since GDAL 2.3
+ */
+CPLJSONObject **CPLJSONObject::GetChildren() const
+{
+    if(nullptr == m_poJsonObject)
+    {
+        return nullptr;
+    }
+    CPLJSONObject **papoChildren = nullptr;
+    size_t nChildrenCount = 0;
+    json_object_iter it;
+    it.key = nullptr;
+    it.val = nullptr;
+    it.entry = nullptr;
+    json_object_object_foreachC( TO_JSONOBJ(m_poJsonObject), it ) {
+        CPLJSONObject *child = new CPLJSONObject(it.key, it.val);
+        papoChildren = reinterpret_cast<CPLJSONObject **>(
+            CPLRealloc( papoChildren,  sizeof(CPLJSONObject *) *
+                        (nChildrenCount + 1) ) );
+        papoChildren[nChildrenCount++] = child;
+    }
+
+    papoChildren = reinterpret_cast<CPLJSONObject **>(
+        CPLRealloc( papoChildren,  sizeof(CPLJSONObject *) *
+                    (nChildrenCount + 1) ) );
+    papoChildren[nChildrenCount] = nullptr;
+
+    return papoChildren;
+}
+
+/**
+ * Get value.
+ * @param  bDefault   Default value.
+ * @return            Boolean value.
+ *
+ * @since GDAL 2.3
+ */
+bool CPLJSONObject::ToBool(bool bDefault) const
+{
+    if( m_poJsonObject /*&& json_object_get_type( TO_JSONOBJ(m_poJsonObject) ) ==
+            json_type_boolean*/ )
+        return json_object_get_boolean( TO_JSONOBJ(m_poJsonObject) ) == 1;
+    return bDefault;
+}
+
+/**
+ * Stringify object to json format.
+ * @param  eFormat Format type,
+ * @return         A string in JSON format.
+ *
+ * @since GDAL 2.3
+ */
+const char* CPLJSONObject::Format(enum PrettyFormat eFormat) const
+{
+    if( m_poJsonObject )
+    {
+        switch ( eFormat ) {
+            case Spaced:
+                return json_object_to_json_string_ext(
+                    TO_JSONOBJ(m_poJsonObject), JSON_C_TO_STRING_SPACED );
+            case Pretty:
+                return json_object_to_json_string_ext(
+                    TO_JSONOBJ(m_poJsonObject), JSON_C_TO_STRING_PRETTY );
+            default:
+                return json_object_to_json_string_ext(
+                    TO_JSONOBJ(m_poJsonObject), JSON_C_TO_STRING_PLAIN);
+        }
+    }
+    return "";
+}
+
+/*! @cond Doxygen_Suppress */
+CPLJSONObject CPLJSONObject::GetObjectByPath(const char *pszPath, char *pszName) const
```

Comment:
Why not using a CPLString& osName instead of C-string manipulation ?

Reply:
In most cases input keys and string values in a CPLJSONObject functions translated into json-c and CPL functions, which accept `const char *`. So this is simple decreasing conversion `const char * -> CPLString -> const char *` operations. But this the only benefit I think. Should I replace `const char* -> const CPLString&` ?